Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp

```diff
@@ -205,6 +212,19 @@ namespace EMStudio
         }
     }
 
+    void AnimViewportWidget::FollowCharacter()
+    {
+        if (m_followCharacter)
+        {
+            AZ::Vector3 camPos = GetViewportContext()->GetCameraTransform().GetTranslation();
+            camPos += m_renderer->GetCharacterCenter() - m_prevCharacterPos;
+            AZ::Transform newCamTransform = GetViewportContext()->GetCameraTransform();
+            newCamTransform.SetTranslation(camPos);
+            GetViewportContext()->SetCameraTransform(newCamTransform);
+        }
+        m_prevCharacterPos = m_renderer->GetCharacterCenter();
```

Comment:
Can you add a comment what this is about with the previous character pos?

Can we still fly around with the camera while in follow mode? Just relative to the character? Or are we locked in then?

Reply:
We are basically adding the delta of the character position change to the camera when the followCharacter is enabled. It is not locking for other camera movement by all means.

Reply:
Ok, can you let that run for some minutes and see if we're drifting here due to floating point precision issues?

Reply:
I let it run a few minutes and didn't notice anything weird.